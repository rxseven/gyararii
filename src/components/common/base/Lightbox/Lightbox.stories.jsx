import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';

import Lightbox from './index';

const title = 'Components/Common/Base/Lightbox';
const props = {
  currentImage: 1,
  images: [
    { src: 'https://via.placeholder.com/480x320?text=Image-001' },
    { src: 'https://via.placeholder.com/480x320?text=Image-002' },
    { src: 'https://via.placeholder.com/480x320?text=Image-003' }
  ],
  isOpen: false,
  onClickNext: action('onClickNext'),
  onClickPrev: action('onClickPrev'),
  onClose: action('onClose')
};

storiesOf(title, module)
  .add('close (default)', () => <Lightbox {...props} />, {
    notes: 'A component is invisible by default (isOpen: false).'
  })
  .add('open', () => <Lightbox {...props} isOpen />);
