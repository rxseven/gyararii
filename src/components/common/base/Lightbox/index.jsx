import PropTypes from 'prop-types';
import React from 'react';
import ReactImages from 'react-images';

const propTypes = {
  currentImage: PropTypes.number.isRequired,
  images: PropTypes.arrayOf(PropTypes.object).isRequired,
  isOpen: PropTypes.bool,
  onClickNext: PropTypes.func.isRequired,
  onClickPrev: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired
};

const defaultProps = {
  isOpen: false
};

const options = {
  backdropClosesModal: true
};

function Lightbox(props) {
  return <ReactImages {...options} {...props} />;
}

Lightbox.propTypes = propTypes;
Lightbox.defaultProps = defaultProps;

export default Lightbox;
