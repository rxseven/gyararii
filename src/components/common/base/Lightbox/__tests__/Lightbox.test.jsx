import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Lightbox from '../index';

describe('<Lightbox />', () => {
  // Arrange
  const props = {
    currentImage: 1,
    images: [{ src: 'image-001' }, { src: 'image-002' }, { src: 'image-003' }],
    isOpen: false,
    onClickNext: jest.fn(),
    onClickPrev: jest.fn(),
    onClose: jest.fn()
  };
  const component = <Lightbox {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
