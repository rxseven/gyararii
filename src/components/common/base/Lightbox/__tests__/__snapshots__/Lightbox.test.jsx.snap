// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Lightbox /> Snapshot tests should render correctly 1`] = `null`;
